src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -445,18 +458,26 @@ private void WriteVerbatimStringLiteral(string literal)
         {
             Write("@\"");
 
-            foreach (char c in literal)
+            // We need to find the index of each '"' (double-quote) to escape it.
+            var start = 0;
+            int end;
+            while ((end = literal.IndexOf('\"', start)) > -1)
```

Comment:
Discussed offline and agreed we should make a change here only if tests show it improves performance. That would  likely happen if the final `string`s are shorter i.e. doubled double-quotes happen less than backslashed characters.